src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableActivityContext.cs

```diff
@@ -61,14 +64,27 @@ internal JToken GetInputAsJson()
         {
             if (this.serializedInput != null && this.parsedJsonInput == null)
             {
-                var objectArray = this.messageDataConverter.Deserialize<object[]>(this.serializedInput);
-
-                if (objectArray?.Length != 1)
+                string actualInput;
+                if (this.inputsAreArrays)
+                {
+                    // The legacy behavior where the DTFx orchestration context schedules the activity results in an input that's
+                    // wrapped in an array. This is unfortunately quite inefficient because it requires us to do an extra serialization
+                    // round-trip in order to correctly interpret the input data.
```

Comment:
This is unfortunately embedded deep into DurableTask.Core, which makes changing it more difficult since it impacts all languages. If we remove the DurableTask.Core dependency from the .NET Isolated SDK, then we can remove this unnecessary overhead (similar to how Java can bypass this stuff).

Reply:
Thanks for the explanation. My 2 cents is that it would be great if we could document this blocker in the code, as I don't think the impact is immediately obvious :)